base/collection_gocb_utils.go

```diff
@@ -9,22 +9,35 @@
 package base
 
 import (
+	"context"
+
 	"github.com/couchbase/gocb/v2"
 	sgbucket "github.com/couchbase/sg-bucket"
 )
 
 // Fills in the GoCB UpsertOptions based on the passed in SG Bucket UpsertOptions
-func fillUpsertOptions(goCBUpsertOptions *gocb.UpsertOptions, upsertOptions *sgbucket.UpsertOptions) {
+func fillUpsertOptions(ctx context.Context, goCBUpsertOptions *gocb.UpsertOptions, upsertOptions *sgbucket.UpsertOptions) {
 	if upsertOptions == nil {
 		return
 	}
 	goCBUpsertOptions.PreserveExpiry = upsertOptions.PreserveExpiry
+	if goCBUpsertOptions.Expiry != 0 && !upsertOptions.PreserveExpiry {
```

Comment:
Shouldn't this be checking for preserveExpiry == true?

Reply:
I think the Info logging also needs to be updated.

Reply:
Whoops, fixed.